src/utils/componentMixin.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+'use strict';
+
+/**
+ * Mixin the attributes and functions to target's prototype
+ * @param {Object} target React component
+ * @param {Object} source the mixin source
+ * @method componentMixin
+ */
+module.exports = function componentMixin(target, source) {
```

Comment:
yeah that’s just a general implementation so that we can keep the original property setting
for our case, we just get it from componentSpec directly
so it’s not used by our case. we can just assign directly
I just keep it as it is an one-time-cost, I don't concern the performance or the complexity of this,  in case we want to use this util for some other usage.